src/ussd_popi_user_data.js

```diff
@@ -420,32 +420,112 @@ go.app = function() {
                     }
                 },
                 next: function(content) {
-                    // TODO: Update phone number on identitystore
-                    return 'state_updated';
+                    return 'state_check_msisdn_available';
+                }
+            });
+        });
+        
+        self.add('state_check_msisdn_available', function(name) {
+            var new_msisdn = utils.normalize_msisdn(self.im.user.answers.state_new_msisdn, '27');
+            self.im.user.set_answer("new_msisdn", new_msisdn);
+
+            return is
+            .list_by_address({msisdn: new_msisdn})
+            .then(function(identities_found) {
+                var identities = (identities_found.results.length > 0) ? identities_found.results : null;
+
+                if (identities === null) {
+                    return self.states.create('state_create_msisdn_change');
+                }
+
+                var opted_out_on_operator = false;
+                var msisdn_available = true;
+                for(i=0; i < identities.length; i++) {
+                    // Msisdn already on this operator but inactive
+                    if (identities[i].id === self.im.user.answers.operator.id) {
+                      if (identities[i].details.addresses.msisdn[new_msisdn].optedout ||
+                              identities[i].details.addresses.msisdn[new_msisdn].inactive)
+                        opted_out_on_operator = true;
+                    }
+                    // Msisdn active on any other than this operator
+                    if (!identities[i].details.addresses.msisdn[new_msisdn].optedout &&
+                            !identities[i].details.addresses.msisdn[new_msisdn].inactive) {
+                        if (identities[i].id !== self.im.user.answers.operator.id) {
+                            msisdn_available = false;
+                            break;
```

Comment:
I'm struggling to get this to work when the code is wrapped in a function so it can be used by the forEach()